### ISSUE TITLE
Replace in search Editor

### ISSUE THREAD

**Issue by YPetremann** [microsoft/vscode#93031]
Having a replace field in the search editor would permit to quickly replace all occurence without having to manage each occurence.

Bonus point for displaying difference like in search panel

**Comment by JacksonKearl** [comment]
Workaround for now is to use the `selectAllSearchEditorMatches` command (Cmd+Shift+L) then alter all at once then use an extension like https://github.com/JacksonKearl/vscode-search-editor-apply-changes to apply those changes to the workspace.

**Comment by JacksonKearl** [comment]
I've been using the aforementioned extension (https://github.com/JacksonKearl/vscode-search-editor-apply-change), and in my view it's quite a bit nicer than a traditional replace box. You can use multicursor to operate on all results at once (select all with Cmd+Shift+L), you can enable context lines to edit surrounding text areas as opposed to just the match result (I for instance yesterday was able to add a call to a function in one place and import that function in another, in a single replace, across dozens of files at once), and you can use editor commands, like `to uppercase`/etc. in your replacing.

It's the most powerful global search experience I know of - I encourage anyone supporting this feature to try out that extension and provide feedback, as any eventual implementation of search editor replace in stable will likely be inspired by that.

**Comment by karlhorky** [comment]
Sounds interesting :) Maybe something very similar can provide the base functionality for the replace feature.

I would request that the feature still retains a quick 3-step workflow (add search query, add replace pattern, run replace) for the more simple find and replace operations though.

**Comment by JacksonKearl** [comment]
> The replace functionality in Search editor should really be disabled if it doesn't actually do anything.

Could you explain? Do you mean to make the editor read-only?

**Comment by swrobel** [comment]
> Do you mean to make the editor read-only?

I would agree that it should be read-only. I find myself accidentally typing it in it all the time and then getting asked if I want to save changes ... changes to what?
<img width="271" alt="Screen Shot 2021-10-11 at 10 52 40 AM" src="https://user-images.githubusercontent.com/157270/136833644-5274efa2-0852-4a6e-8abb-e97a2532a58b.png">

**Comment by bluikko** [comment]
> Could you explain? Do you mean to make the editor read-only?

I had a large project-wide change. I found the search editor and was delighted, starting to use replace over a few hours to complete the change to hundreds of files. At the end the delight quickly disappeared when there was no way to save all that work... it is totally counter-intuitive currently.

**Comment by ArturoDent** [comment]
@bluikko Did you read above about https://github.com/JacksonKearl/vscode-search-editor-apply-changes?  It can save your search editor changes to all those files.

**Comment by laizp** [comment]
any progress? : )

**Comment by karlhorky** [comment]
To @andreamah (or anyone else working on this), just wanted to add that still in 2023, this feature is an almost daily wish of mine when I'm searching in VS Code.

The Search Editor was a great feature and brought VS Code almost to par with Sublime Text, but just needs this search and replace. Would be great to get [a simple 3-step find and replace](https://github.com/microsoft/vscode/issues/93031#issuecomment-607456479) in the Search editor 🙌

If there could also be a followup project to enable something more advanced like @JacksonKearl's [Search Editor: Apply Changes](https://marketplace.visualstudio.com/items?itemName=jakearl.search-editor-apply-changes) ([repo](https://github.com/JacksonKearl/vscode-search-editor-apply-changes)), which was also mentioned a few times in this thread, that would also be cool, but not essential for a first step.

**Comment by gene-pavlovsky** [comment]
I'd like this too, especially would be useful when using RegExp search and reference the matched groups in the replace string

**Comment by DougPlumley** [comment]
I stumbled across this as well because I thought the search editor was an awesome tool until I realized I couldn't apply the changes against the source files.

**Comment by TheGoldMonkey** [comment]
Can this be reprioritized?

**Comment by devuxer** [comment]
> Can we make the search "editor" read-only while this is being worked on?
> 
> I think it is confusing to be able to edit and not cause any effect in the files (more confusing that not being able to edit in the first place).

As mentioned in the [second comment](https://github.com/microsoft/vscode/issues/93031#issuecomment-603527524) of this issue, there is an [extension](https://marketplace.visualstudio.com/items?itemName=jakearl.search-editor-apply-changes) that allows us to save the changes we make in the search editor, so definitely do not remove the editing capability!

That said, as long as the search editor exists, the ability to save changes really should be a first-class feature of VSCode. The fact that this can be done with an extension proves that the feature is feasible.

**Comment by sheldonhull** [comment]
I dug around to find this feature request.
For years I've wondered about how to save my changes in the search panel.
With the lines of context, it's a fantastic way to control some manual edits across multiple files.
Would love to see this get some attention.
